src/components/questionnaire/QuestionnaireList.tsx

```diff
@@ -1,30 +1,45 @@
 import { useNavigation } from '@react-navigation/native';
 import React from 'react';
-import { StyleSheet, View } from 'react-native';
+import { Alert, Platform, StyleSheet, ToastAndroid, View } from 'react-native';
+import { getAllQuestionnaireDataByCode } from '../../data/api/Questionnaire';
+import { questionnaire } from '../../data/localStorage/ParticipantCode';
 import { Questionnaire } from '../../models/Questionnaire';
 import Button from '../generic/Button';
 
 interface QuestionnaireListProps {
-  questionnaires: Questionnaire[];
+  questionnaires: questionnaire[];
 }
 
 const QuestionnaireList = ({ questionnaires }: QuestionnaireListProps) => {
   const navigation = useNavigation();
 
+  const handleQuestionnaireSelected = async (code: string) => {
+    let questionnaireResponse = await getAllQuestionnaireDataByCode(code);
+    if (questionnaireResponse.status === 200) {
+      const questionnaireData: Questionnaire = questionnaireResponse.data;
+      // @ts-ignore next-line
+      navigation.navigate('Questionnaire', {
+        title: questionnaireData.title,
+        questionnaire: questionnaireData,
+      });
+    } else {
+      const msg = 'Vragenlijst niet gelukt op te halen';
+      if (Platform.OS === 'android') {
+        ToastAndroid.show(msg, ToastAndroid.LONG);
+      } else {
+        Alert.alert(msg);
+      }
+    }
+  };
   return (
     <>
-      {questionnaires?.map((questionnaire, index) => {
+      {questionnaires?.map((questionnaireItem, index) => {
         return (
           <View key={index} style={styles.buttonView}>
             <Button
-              title={questionnaire.title}
-              onButtonPress={() => {
-                // @ts-ignore next-line
-                navigation.navigate('Questionnaire', {
-                  title: questionnaire.title,
-                  questionnaire: questionnaire,
-                });
-              }}
+              accLabel={questionnaireItem.name}
+              title={questionnaireItem.name}
+              onButtonPress={() => handleQuestionnaireSelected(questionnaireItem.code)}
```

Comment:
Hadden we niet bedacht om de questionnaire te cachen of gingen we dit niet meer doen?

Reply:
Ja, wel bedacht, maar nog geen prio

Reply:
Oke is goed!